### PR TITLE
Create namespace and enable for AWS, Azure and KVM

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#1902

Implements creating the giantswarm namespace and adds it to all 3 providers.

### CODE REVIEW

File: pkg/v3/resource/namespace/update.go [hunk 1]
```diff
@@ -3,13 +3,36 @@ package namespace
...
 )
 
+// ApplyUpdateChange is a no-op because the namespace in the guest cluster is
+// not updated.
 func (r *Resource) ApplyUpdateChange(ctx context.Context, obj, updateChange interface{}) error {
 	return nil
 }
 
 func (r *Resource) NewUpdatePatch(ctx context.Context, obj, currentState, desiredState interface{}) (*controller.Patch, error) {
+	create, err := r.newCreateChange(ctx, obj, currentState, desiredState)
```
Comment (indyfree) [inline pkg/v3/resource/namespace/update.go]: Just out of interest: Why do we need to implement `NewUpdatePatch` if we don't update the namespace?

Comment (rossf7) [reply pkg/v3/resource/namespace/update.go]: We can't reliably distinguish between create and update events from the informer. So operatorkit treats both the same. This is why there is NewUpdatePatch but no NewCreatePatch.

**Comment by rossf7** [comment]
@indyfree The name of the namespace is a constant in resource.go.

https://github.com/giantswarm/cluster-operator/blob/46d00563822d08bf6139710b9597883e3722e4ba/pkg/v3/resource/namespace/resource.go#L23

File: pkg/v3/resource/namespace/update.go (hunk 1 above)
Comment (kopiczko) [reply pkg/v3/resource/namespace/update.go]: This is to make sure things are idempotent. You can think of it as `CreateOrUpdate`.

**Comment by rossf7** [comment]
Due to the API whitelisting we now connect to the master service used by Prometheus rather than the public Route 53 domain. 

Tested on ginger. Going with this.

**Pull request merged**